File: src/data/query.rs
use crate::data::format::format_result;

pub fn run_query(sql: &str) -> String {
    let raw = format!("result_{}", sql);
    format_result(&raw)
}

pub fn build_query(table: &str) -> String {
    format!("SELECT * FROM {}", table)
}

File: src/data/query.ts

import { formatResult } from './format';
import { getCached } from './cache';

export function runQuery(sql: string) {
  const cached = getCached(sql);
  if (cached) return cached;
  return formatResult({ sql, rows: [] });
}

export function buildQuery(table: string, conditions: any) {
  return 'SELECT * FROM ' + table;
}
